store/src/main/scala/geotrellis/store/file/FileCollectionLayerReader.scala

```diff
@@ -68,7 +73,7 @@ class FileCollectionLayerReader(
 
 object FileCollectionLayerReader {
   def apply(attributeStore: AttributeStore, catalogPath: String): FileCollectionLayerReader =
-    new FileCollectionLayerReader(attributeStore, catalogPath)
+    new FileCollectionLayerReader(attributeStore, catalogPath, BlockingThreadPool.executionContext)
```

Comment:
Are you avoiding assigning `BlockingThreadPool.executionContext` as default value here intentionally? If so what is the rationale?

Reply:
@echeipesh it is a default here; what do you mean?

Reply:
Oh, I mean about the `FileCollectionLayerReader` constructor rather than the `apply`.

Reply:
@echeipesh oh, a great question! my answer would be that I didn't think about it too much and was worrying mostly about serialization and changing constructors / etc minimally since it's already a significant PR; But I see what you're saying. We probably need to review all IOReaders / Writers constructors and apply methods. Don't know why does it leave in the companion object actually.

Reply:
This is the only one I've spotted so far

Reply:
If its the only one then it is an oversight :+1: